Source/JavaScriptCore/inspector/agents/InspectorRuntimeAgent.cpp

```diff
@@ -203,15 +200,15 @@ Protocol::ErrorStringOr<std::tuple<Ref<Protocol::Runtime::RemoteObject>, std::op
         muteConsole();
     }
 
-    injectedScript.callFunctionOn(errorString, objectId, functionDeclaration, arguments ? arguments->toJSONString() : nullString(), returnByValue.value_or(false), generatePreview.value_or(false), result, wasThrown);
+    injectedScript.callFunctionOn(objectId, functionDeclaration, arguments ? arguments->toJSONString() : nullString(), returnByValue.value_or(false), generatePreview.value_or(false), awaitPromise.value_or(false), [callback = WTFMove(callback)] (Protocol::ErrorString& errorString, RefPtr<Protocol::Runtime::RemoteObject>&& result, std::optional<bool>&& wasThrown, std::optional<int>&&) {
+        if (!result)
+            callback->sendFailure(errorString);
+        else
+            callback->sendSuccess(result.releaseNonNull(), WTFMove(wasThrown));
+    });
 
     if (pauseAndMute)
```

Comment:
should this also move inside the lambda above?

or is that specifically not desired since other async things could happen in between whatever was `eval` and the `.then()`?

Reply:
It is the latter. I don't think it is safe to mute console and skip exceptions for arbitrarily long time until the function is resolved.